source/connection-monitoring-and-pooling/tests/wait-queue-fairness.yml

```diff
@@ -1,31 +1,39 @@
 version: 1
 style: unit
-description: must issue Connections to threads in the order that the threads eneterd the queue
+description: must issue Connections to threads in the order that the threads entered the queue
 poolOptions:
   maxPoolSize: 1
+  waitQueueTimeoutMS: 1000
 operations:
     # Check out sole connection in pool
   - name: checkOut
     label: conn0
     # Create 4 threads, have them all queue up for connections
     # Note: this might become non-deterministic depending on how you
     # implement your test runner. The goal is for each thread to
-    # have started and begun checkOut before the next thread starts
+    # have started and begun checkOut before the next thread starts.
+    # The sleep operations should make this more consistent.
   - name: start
     target: thread1
   - name: checkOut
     thread: thread1
     label: conn1
+  - name: wait
+    ms: 10
   - name: start
     target: thread2
   - name: checkOut
     thread: thread2
     label: conn2
+  - name: wait
+    ms: 20
```

Comment:
Is it necessary for the `ms` count to increase for each one?

Reply:
The idea is that if all of the threads are running concurrently (which will happen in languages without a GIL), there will be a race if they all sleep for the same amount of time, since they will wake up roughly the same time. By increasing each sleep, we can ensure that the threads wake up in the order we want.

Reply:
Ah. In that case, you need to add the `thread` to each of these wait calls. Currently, all these `wait`s are on the main thread.

Reply:
I think this still works; the later threads aren't started until after a sleep, meaning they won't attempt to check out a connection until after the previous one already has had a chance to enter the wait queue.

Reply:
it will definitely still work, but if we are sleeping on the main thread instead of individual threads, they might as well only be 10ms per sleep

Reply:
Ah, I see what you're saying; I'll update it